Add flag to temporarily disable MKL-DNN conv

jamesr66a opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23718 [wip][quantization] Enable OSS quantization tests
* **#23837 Add flag to temporarily disable MKL-DNN conv**

This is a temporary workaround to an issue in MKL-DNN's Convolution backwards implementation: https://github.com/pytorch/pytorch/issues/23825

It is only used to enable testing quantization

Differential Revision: [D16659081](https://our.internmc.facebook.com/intern/diff/D16659081)

Commit:
Add flag to temporarily disable MKL-DNN conv

Commit:
Update on "Add flag to temporarily disable MKL-DNN conv"

Commit:
Update on "Add flag to temporarily disable MKL-DNN conv"

bddppq commented on test/test_qat.py:
```diff
@@ -9,13 +9,29 @@
+from contextlib import contextmanager
+
+@contextmanager
+def disable_mkldnn_conv():
```
move this to torch/utils/mkldnn.py?

bddppq commented on torch/csrc/utils/init.cpp:
```diff
@@ -2,6 +2,8 @@
 #include <torch/csrc/utils/init.h>
 #include <torch/csrc/utils/throughput_benchmark.h>
 
+#include <ATen/native/Convolution.h>
```
Could you add a comment here to say this is a temporary workaround? (Or is there is better place to add the binding? because technically jit shouldn't need to depend on conv).

jamesr66a replied:
This isn't in a jit folder though?

bddppq reviewed (approved):
Overall lgtm

Commit:
Update on "Add flag to temporarily disable MKL-DNN conv"


This is a temporary workaround to an issue in MKL-DNN's Convolution backwards implementation: https://github.com/pytorch/pytorch/issues/23825

It is only used to enable testing quantization

Differential Revision: [D16639467](https://our.internmc.facebook.com/intern/diff/D16639467)

Commit:
Update on "Add flag to temporarily disable MKL-DNN conv"


This is a temporary workaround to an issue in MKL-DNN's Convolution backwards implementation: https://github.com/pytorch/pytorch/issues/23825

It is only used to enable testing quantization

Differential Revision: [D16659081](https://our.internmc.facebook.com/intern/diff/D16659081)

fmassa commented on torch/utils/mkldnn.py:
```diff
@@ -150,3 +150,19 @@ def m_fn_rec(m):
+from contextlib import contextmanager
+
+@contextmanager
+def disable_mkldnn_conv():
```
Apparently the bug in MKLDNN has been fixed in the latest release
